lib/user_space/user_x86_64_atomic.h

```diff
@@ -114,13 +162,33 @@ static inline void m0_atomic64_sub(struct m0_atomic64 *a, int64_t num)
 static inline int64_t m0_atomic64_add_return(struct m0_atomic64 *a,
 						  int64_t delta)
 {
+#ifdef CONFIG_X86_64
 	long result;
 
 	result = delta;
 	asm volatile("lock xaddq %0, %1;"
 		     : "+r" (delta), "+m" (a->a_value)
 		     : : "memory");
 	return delta + result;
+#elif defined CONFIG_AARCH64
+	int64_t		result;
+	uint64_t	tmp;
+
+	asm volatile("// atomic64_add_return \n"	\
+		     "  prfm    pstl1strm, %2\n"	\
+		     "1:ldxr    %0, %2\n"		\
+		     "  add     %0, %0, %3\n"   	\
+		     "  stlxr   %w1, %0, %2\n"  	\
+		     "  cbnz    %w1, 1b\n"		\
+		     "  dmb ish"			\
+		     : "=&r" (result), "=&r" (tmp), "+Q" (a->a_value) \
+		     : "Ir" (delta)  			\
+		     : "memory");
+	return result;
+#else
+#error  "The Platform is not supported"
+#endif
+	return 0;
```

Comment:
Needless return.

Reply:
it is required, as it  solves some hare  compilation issue

Reply:
Which issue?

Reply:
there are compilation error that non-void function does not return anthing

Reply:
Does it still complains? On which platform?

Reply:
yes, i observed in aarch64 platform

Reply:
But there is a return statement for aarch64 platform in this function, isn't there? (Just a few lines above.)

Reply:
yea, Since they are inside the macro , the  compiler sometime does not infer , if not put outside the macros

Reply:
That's weird. Can you try to compile it now and paste the error here please?

Reply:
This is not relevant anymore.